multiplexer/main.go

```diff
@@ -13,33 +19,135 @@ type URLs struct {
 	List []string `json:"list"`
 }
 
+//Client is a http clieng
+type Client struct {
+	client *http.Client
+}
+
+//Response is the response sent to the client
+type Response struct {
+	index   int
+	content []byte
+}
+
+//Job is a single url with its index
+type Job struct {
+	index int
+	site  string
+}
+
+func newClient() *Client {
+	return &Client{
+		client: &http.Client{
+			Timeout: time.Second,
+		},
+	}
+}
+
+func (c Client) getContent(url string) ([]byte, error) {
+	resp, err := c.client.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return b, nil
+}
+
 func main() {
 	var urls URLs
+	var jwg sync.WaitGroup
+	var rwg sync.WaitGroup
+	jobs := make(chan Job)
+	result := make(chan Response)
+	workers := 5
 
-	client := http.Client{
-		Timeout: time.Second,
-	}
+	done := make(chan os.Signal, 1)
+	signal.Notify(done, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+
+	client := newClient()
 
 	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
 		if err := json.NewDecoder(r.Body).Decode(&urls); err != nil {
 			return
 		}
 
-		for _, payLoad := range urls.List {
-			resp, err := client.Get(payLoad)
-			if err != nil {
-				w.Write([]byte(err.Error()))
+		if len(urls.List) > 20 {
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Write([]byte("amount of url's exceeded limit"))
+			return
+		}
+
+		//reciving result, sorting, and then sending to client
+		rwg.Add(1)
+		go func() {
+			responses := make([]string, len(urls.List))
+			for r := range result {
+				responses[r.index] = string(r.content)
 			}
-			defer resp.Body.Close()
 
-			b, err := ioutil.ReadAll(resp.Body)
-			if err != nil {
-				w.Write([]byte(err.Error()))
+			for i, content := range responses {
```

Comment:
You forgot to sort your response. You can't use map here, you need to sort your `[]Response` for example

Reply:
i did that on line 91. putting the content of the site in the same index of the site as it was sent